### ISSUE TITLE
Gradient estimator methods like straight through estimators for gradient approximation of non-differentiable landscapes of a model's graph.

### ISSUE THREAD

**Issue by Puranjay-del-Mishra** [pytorch/pytorch#72772]
### 🚀 The feature, motivation and pitch

I'm working on a model in which I'm quantizing the weights before the forward propogation. This makes the model's landscape non-differentiable (gradients becoming 0). Regular gradient descent algorithms don't work  here and thus gradient estimator methods are needed to converge the model.
 

### Alternatives

Changing the computational graph or reworking the backward calls for all the layers with quantized weights (or user specified layers) should work. However i was not able to make this work.

cc @ezyang @albanD @zou3519 @gqchen @pearu @nikitaved @soulitzer @Lezcano @Varal7

**Comment by soulitzer** [comment]
> Changing the computational graph or reworking the backward calls for all the layers with quantized weights (or user specified layers) should work. 

Yes, a custom autograd Function should work here. If you have any issues with that, feel free to post on the [forum](https://discuss.pytorch.org/) and link here.
